### ISSUE TITLE
inkyphat.show() is very slow

### ISSUE THREAD

**Issue by gorx85** [pimoroni/inky-phat#25]
The method inkyphat.show() is unnecessarily slow. On my Pi Zero the following code takes almost 2 seconds to complete:

```
    for y in range(WIDTH):
        for x in range(HEIGHT):
            _panel.set_pixel(x, y, _image.getpixel((y, x)))
```

I suggest to use numpy to set the whole of `_panel`'s buffer at once, instead of setting each pixel individually. In order to transform `_image` to the required buffer, one could do something along the lines of this:

```
    buffer = numpy.swapaxes(numpy.array(_image, dtype=numpy.uint8),0,1)
```

This would save almost 2 seconds for each display refresh.

Of course this should be adapted a bit in order to avoid copying pixels which are not part of the palette, but the basic idea should be clear.

**Comment by Gadgetoid** [comment]
I'm planning to all but depricate this library in favour of its replacement: Inky - https://github.com/pimoroni/inky/

Of course, the problem you mention is still present however it's been moved from `show()`, which now handles an agnostic numpy back-buffer, to the `set_image` method which is also stepping through pixels in basically the same way as this library:

https://github.com/pimoroni/inky/blob/dbd6b15049ef1727915c172b7ceab1071d73c96e/library/inky/inky.py#L227-L241

In fact, this is compounded when using a 400x300 pixel resolution display.

From what I can tell I could rely entirely upon numpy to handle these cases (list, PIL image or numpy-based image):

```python
def set_image(self, image):
    self.buf = numpy.array(image, dtype=numpy.uint8).reshape((self.width self.height)).swapaxes(1, 0)
```

So, in short, thank you for pointing this out!
